### PR TITLE
Make search box display configurable.

### LINKED ISSUE #46: 8.x Make display of search form optional
Allow the search form to be hidden for sites for which it is not necessary.

### PR DESCRIPTION
**Pull request by pyrello**
Makes displaying the search box optional and configurable via the settings form.

resolves #46

### CODE REVIEW

**Comment by pyrello** [comment]
FYI, I have not made any updates to the CSS as part of this change. If you think CSS changes are necessary, let me know and I'll check it out.

**Comment by joewhitsitt** [comment]
I think the use-case is there, so we will pass this along to OSC. I'd love to coordinate this with any HDS work/refactoring #36  so that we only have to touch CSS once.

File: config/install/uiowa_bar.settings.yml [hunk 1]
```diff
@@ -1,4 +1,5 @@
 uiowa_bar:
   wordmark: uiowa
+  display_search: true
```
Comment (richardbporter) [inline config/install/uiowa_bar.settings.yml]: I think this will be an integer (0/1) without the `#return_value` set on the checkbox form element.

**Review by richardbporter — CHANGES_REQUESTED** [review]
SiteNow sites seem to display fine without the search box present. However, we'll need an update hook to create the new config key and set it to true (1). Otherwise, it will no longer be displayed for existing sites.

**Comment by pyrello** [comment]
I tested this against https://github.com/uiowa/uiowa/pull/916 on my local machine by running `blt dsa`, which runs updates. The update applied without errors on default, and careers.uiowa.edu. I attempted to take a look at careers.uiowa.edu to see if the setting was as expected, but I was not able to log into the site by running `drush @careers.local uli`. It kept giving me a message that the login link was expired.

File: config/install/uiowa_bar.settings.yml (hunk 1 above)
Comment (pyrello) [reply config/install/uiowa_bar.settings.yml]: Changed to 1 by default

**Comment by richardbporter** [comment]
I couldn't figure out why the update hook wasn't working for me locally. Turns out this module configuration is nested under an extra `uiowa_bar` key which is odd. I'm not going to bother changing it given a lot more will change with this module in the future.

**Comment by pyrello** [comment]
Sounds good. Thanks.

**Review by richardbporter — APPROVED** [review]

**Comment by richardbporter** [comment]
Tested with and without links. Displays fine on mobile/desktop.

**Pull request merged**